biothings/web/handlers/query.py

```diff
@@ -149,10 +150,14 @@ async def get(self):
                 if field.startswith('_'):
                     meta.pop(field, None)
 
-        meta = self.extras(meta)  # override here
+        if (iscoroutinefunction(self.extras)):
+            meta = await self.extras(meta)
+        else:
+            meta = self.extras(meta)
+
         self.finish(dict(sorted(meta.items())))
 
-    def extras(self, _meta):
+    async def extras(self, _meta):
```

Comment:
@everaldorodrigo if `extra` can be sync or async, shall we need the original method unchanged as a sync method? Or is there any reason to make it an async method by default?

Reply:
> @everaldorodrigo if `extra` can be sync or async, shall we need the original method unchanged as a sync method? Or is there any reason to make it an async method by default?

@newgene, there's no strong reason in making it async by default. We could keep it sync.

Reply:
👍 let's keep it unchanged then, then the subclass can override it as async method if needed.

Reply:
:done: :bowtie: